### PR TITLE
Update golang SDK with new API changes.

### PR DESCRIPTION
**Pull request by mplooster**
Also cleaned up some things by running `go mod tidy`. The `diatheke.pb.go` file was autogenerated, and does not need to be reviewed. Documentation will be updated later - after other languages have also been updated with the API changes - so it is not included in this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Code generation has the following dependencies:
   - The static website generator (hugo)
 
 A few system dependencies are required:
-  - Go >= 1.12
+  - Go >= 1.14
```
Comment (jhollowayj) [inline README.md:30]: Should we jump to 1.15?

File: examples/go/cmd/demo-audio/main.go [hunk 2]
```diff
@@ -198,8 +205,7 @@ func (manager *sessionManager) ToggleOff() {
 // StartEventStream creates a new event stream for the session being
 // managed.
 func (manager *sessionManager) StartEventStream() {
-	// Start the event stream on a separate go routine
-	go manager.handleEvents()
+	manager.handleEvents()
```
Comment (jhollowayj) [inline examples/go/cmd/demo-audio/main.go:208]: I assume this change is fine.  I didn't see any changes on manager.handleEvents() to start a new go routine, but I did see other places where the for loop was wrapped in a goroutine.

File: examples/go/cmd/demo-cli/main.go [hunk 3]
```diff
@@ -155,36 +160,39 @@ func handleEvents(session diatheke.Session) {
...
+
+			// This indicates the stream has finished, which will happen
+			// when the session ends or the context used to create the stream
+			// closes.
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				textui.PrintError("Event stream error: %v\n", err)
+				return
```
Comment (jhollowayj) [inline examples/go/cmd/demo-cli/main.go:177]: Is there a reason to break and return differently?  Should we make these the same?

File: examples/go/cmd/demo-audio/eventStream.go [hunk 4]
```diff
@@ -33,36 +33,39 @@ func (manager *sessionManager) handleEvents() {
...
+
+			// This indicates the stream has finished, which will happen
+			// when the session ends or the context used to create the stream
+			// closes.
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				textui.PrintError("Event stream error: %v\n", err)
+				return
```
Comment (jhollowayj) [inline examples/go/cmd/demo-audio/eventStream.go:50]: can we `break` on both or `return` on both?

File: README.md (hunk 1 above)
Comment (mplooster) [reply README.md:30]: Our SDK generator image that I used for the update is still on 1.14. We probably should update it eventually, but I didn't want to slow down this release for that update.

File: examples/go/cmd/demo-audio/eventStream.go (hunk 4 above)
Comment (mplooster) [reply examples/go/cmd/demo-audio/eventStream.go:50]: Sure

File: examples/go/cmd/demo-audio/main.go (hunk 2 above)
Comment (mplooster) [reply examples/go/cmd/demo-audio/main.go:208]: You are correct. The go routine was moved to be around just the for loop.

File: examples/go/cmd/demo-cli/main.go (hunk 3 above)
Comment (mplooster) [reply examples/go/cmd/demo-cli/main.go:177]: There was a nuanced difference before, but now that it is wrapped in the go routine, it doesn't matter. I can make them the same.

**Pull request merged**